src/providers/eveonline.ts

```diff
@@ -0,0 +1,37 @@
+import { Profile } from ".."
+import { OAuthConfig, OAuthUserConfig } from "./oauth"
+
+export interface EVEOnlineProfile extends Profile {
+  CharacterID: number;
+  CharacterName: string;
+  ExpiresOn: string;
+  Scopes: string;
+  TokenType: string;
+  CharacterOwnerHash: string;
+  IntellectualProperty: string;
+}
+
+export default function EVEOnline<P extends Record<string, any> = EVEOnlineProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "eveonline",
+    name: "EVE Online",
+    type: "oauth",
+    token: "https://login.eveonline.com/oauth/token",
+    userinfo: "https://login.eveonline.com/oauth/verify",
+    authorization: {
+      params: { scope: "publicData" },
+      url: "https://login.eveonline.com/oauth/authorize",
+    },
+    profile(profile: P) {
```

Comment:
```suggestion
    profile(profile) {
```

Type here is supposed be inferred.

Reply:
Sure. Should update the Google provider as well, that's where I pulled the base from.